src/main/java/org/jenkinsci/test/acceptance/docker/Docker.java

```diff
@@ -158,6 +122,80 @@ public DockerImage build(Class<? extends DockerContainer> fixture) throws IOExce
         }
     }
 
+    //package scope for testing purposes. Ideally we should encapsulate Docker interactions so they can be mocked
+    // and call public method.
+    void copyDockerfileDirectory(Class<? extends DockerContainer> fixture, DockerFixture f, File dir)
+            throws IOException {
+        String dockerfilePath = resolveDockerfileLocation(fixture, f);
+        copyDockerfileDirectoryFromClasspath(fixture, dockerfilePath, dir);
+    }
+
+    private String resolveDockerfileLocation(Class<? extends DockerContainer> fixture, DockerFixture f) {
+        String prefix = null;
+        if(isSpecificDockerfileLocationSet(f)) {
+            prefix = f.dockerfile().replace('.', '/').replace('$', '/');
+        } else {
+            prefix = fixture.getName().replace('.', '/').replace('$', '/');
+        }
+        return prefix;
+    }
+
+    private void copyDockerfileDirectoryFromClasspath(Class<? extends DockerContainer> fixture, String dockerfileLocation, File dir) throws IOException {
+        File jar = null;
+        try {
+            jar = Which.jarFile(fixture);
+        } catch (IllegalArgumentException e) {
+            // fall through
```

Comment:
Yes in separate commit/PR make sense to not mix things. To log it you mean using Sysout directly?